projects/com.oracle.truffle.llvm.tools/src/com/oracle/truffle/llvm/tools/GCC.java

```diff
@@ -30,34 +30,58 @@
 package com.oracle.truffle.llvm.tools;
 
 import java.io.File;
-import java.io.IOException;
 
 import com.oracle.truffle.llvm.runtime.options.LLVMBaseOptionFacade;
 import com.oracle.truffle.llvm.tools.util.ProcessUtil;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+public final class GCC {
+
+    private static final File PROJECT_ROOT = new File(LLVMBaseOptionFacade.getProjectRoot() + File.separator + LLVMTools.class.getPackage().getName());
+    private static final File TOOLS_ROOT = new File(PROJECT_ROOT, "tools");
+
+    private static final File LLVM_DRAGONEGG = new File(TOOLS_ROOT, "/dragonegg/dragonegg-3.2.src/dragonegg.so");
+    private static final String[] GCC_SUPPORTED_VERSIONS = {"4.6", "4.7", "4.8"};
 
-public final class GCC extends CompilerBase {
+    public enum Tool {
+        GPP_PATH("g++"),
+        GFORTRAN_PATH("gfortran"),
+        GCC_PATH("gcc");
 
-    private static final File GPP_PATH = Mx.executeGetGCCProgramPath("g++");
-    private static final File GFORTRAN_PATH = Mx.executeGetGCCProgramPath("gfortran");
-    private static final File GCC_PATH = Mx.executeGetGCCProgramPath("gcc");
+        private final Path toolPath;
+
+        Tool(String programName) {
+            this.toolPath = findTool(programName);
+        }
+
+        private static Path findTool(String toolName) {
+            Path path = ToolLookup.findTool(toolName + "(-mp.*)?", s -> Arrays.stream(GCC_SUPPORTED_VERSIONS).filter(v -> s.contains(v)).findAny().isPresent());
+            if (path == null) {
+                throw new IllegalStateException("Sorry, we could not find a " + toolName);
```

Comment:
Could we have an informative error message here which tells the user what he/she has to do to proceed? For example, we can output that a GCC version out of `GCC_SUPPORTED_VERSIONS` needs to be installed.